Middleware/src/main/java/ca/concordia/ngci/tools4cities/middleware/consumers/SortedEnergyConsumptionConsumer.java

```diff
@@ -0,0 +1,83 @@
+package ca.concordia.ngci.tools4cities.middleware.consumers;
+
+import java.util.*;
+
+import ca.concordia.ngci.tools4cities.middleware.middleware.AbstractConsumer;
+import ca.concordia.ngci.tools4cities.middleware.middleware.IConsumer;
+import ca.concordia.ngci.tools4cities.middleware.middleware.IProducer;
+
+public class SortedEnergyConsumptionConsumer extends AbstractConsumer<String> implements IConsumer<String> {
+    private List<String> results;
+    private Map<Integer, Double> monthlyConsumption;
+
+    public SortedEnergyConsumptionConsumer(Set<IProducer<String>> producers) {
+        super(producers);
+        this.results = new ArrayList<>();
+        this.monthlyConsumption = new HashMap<>();
+    }
+
+    @Override
+    public List<String> getResults() {
+        return results;
+    }
+
+    @Override
+    public void newDataAvailable(List<String> data) {
+    	if (data == null || data.isEmpty()) {
+            System.err.println("Received null or empty data!");
+            return;
+        }
+
+        // Process the incoming data
+        processData(data);
+    }
+
+    private void processData(List<String> data) {
+    	// Skip header by starting from the second line if the first line is a header
+        boolean isHeader = data.get(0).contains("kWh"); // Adjust this based on header content
+        int startIndex = isHeader ? 1 : 0;
+
+        for (int i = startIndex; i < data.size(); i++) {
+            String line = data.get(i);
+            String[] values = line.split(",");
+
+            // Check for valid line length
+            if (values.length < 4) {
+                System.err.println("Invalid line: " + line);
+                continue;
+            }
+
+            String date = values[2];
+            double kWh;
+            
+            // Parse kWh value
+            try {
+                kWh = Double.parseDouble(values[3]);
+            } catch (NumberFormatException e) {
+                continue;
+            }
+
+            // Extract the month from the date (assuming format YYYY-MM-DD)
+            int month = Integer.parseInt(date.substring(5, 7));
+
+            // Sum the kWh values for each month
+            monthlyConsumption.put(month, monthlyConsumption.getOrDefault(month, 0.0) + kWh);
```

Comment:
When I run the tests here, I get a NullPointerException on monthlyConsumption. That happens because the constructor is asynchronous and we cannot guarantee it will initialize the attribute before this method gets called. My suggestion: 
Map<Integer, Double> consumptionFromCsv = new HashMap<>();
consumptionFromCsv.put(month, consumptionFromCsv.getOrDefault(month, 0.0) + kWh);
setMonthlyConsumption(consumptionFromCsv);